js/source/worker.js

```diff
@@ -35,14 +61,28 @@ function Worker(self) {
 }
 
 util.extend(Worker.prototype, {
-    'set layers': function(layers) {
-        this.layers = {};
-        var that = this;
+    'set style': function(mapId, style) {
+        var key = createKeyForStyle(style);
+
+        var prevKey = this.styles[mapId];
+        if (prevKey === key) return;
+        this.styles[mapId] = key;
+
+        var shouldDeletePreviousStyle = util.values(this.styles)
+            .some(function (usedKey) { return usedKey === prevKey; });
+        if (shouldDeletePreviousStyle) {
+            delete this.layers[prevKey];
+            delete this.layerFamilies[prevKey];
+        }
+
+        if (this.layers[key]) return;
+
+        var styleLayers = this.layers[key] = {};
```

Comment:
In this change, `this.layers` and `this.layerFamilies` gain a layer of indirection, becoming maps from a 'style key' to the style's set of layers and layer families.  This allows a Worker to track multiple map instances' styles -- but by keying them off of a 'style key' rather than a map instance id, we allow for two map instances with the same style to share work.